app/models/reservation.rb

```diff
@@ -36,6 +35,11 @@ class Reservation < ActiveRecord::Base
   delegate :lock_window, to: :product, prefix: true
   delegate :owner, to: :account, allow_nil: true
 
+  def canceled?
+    return nil unless order_detail
```

Comment:
I went with returning `nil` because that duplicates the behavior of `delegate` with `allow_nil: true`, which is what I was originally doing before I had to write a custom method due to the name change. I'm fine with changing it to false, though.